shells/pipes-shell/pipe.js

```diff
@@ -106,10 +133,21 @@ const instantiateRecipe = async (arc, manifest, name) => {
   }
 };
 
+const oneChange = (store, callback, arc) => {
```

Comment:
It's easy to mis-read `oneChange` and `onChange`, could be worth a more differentiated name for one or the other.

Reply:
Yes, I saw that and I should have fixed it. 

Fwiw, this code is going to undergo a fair amount of refactoring in the near future and I instinctively don't spend time on things I know will be better abstracted soon.